### PR TITLE
DataSeeder Upgrade

### PR DESCRIPTION
**Pull request by Flemingjp**
~~_**PR in draft as is dependent on #83. The relevant commits are [here](https://github.com/biobankinguk/biobankinguk/pull/90/files/41b763a2eb948da11c39c919e4b772e43e120a96..cef8228b01a9f0a6c4c4e265f02f420907f61c31)**_~~

Upgraded the DataSeeder to update csv's of harmonized entities, and to include seeding for the newly introduced API entities.

Other upgrades include

- The `DbSet` for a given entity is now inferred directly from the generic `db.Set<TEntity>()` vs `db.Set(Type type)` removing the need for a lookup table of type and table name
- The loaded csv data is also uses `csv.GetRecords<T>()` vs `csv.GetRecords(Type type)` for the same reasons
- Unified all seed actions into one list, with the seeding order determined by the list order. This removes the need for the separate `LeaveTilLast` list of seed actions. 
- `SeedingService` now uses the logger vs `Console.WriteLine` and all logging is done via the `Seed<T>` method such that all seeding is properly logged
- Updated the sample-seed-data to reflect the changes to the entities fields and names

### CODE REVIEW

File: src/Entities/Api/ReferenceData/OntologyVersion.cs [hunk 1]
```diff
@@ -27,7 +27,6 @@ public class OntologyVersion
         /// <summary>
         /// Ontology to which the version relates.
         /// </summary>
-        [Required]
-        public Ontology Ontology { get; set; }
```
Comment (Flemingjp) [inline src/Entities/Api/ReferenceData/OntologyVersion.cs:31]: Removal of `[Required]` as would throw an error when trying to insert any entity into the table with a `null` value - not sure of the implications to the API

Comment (beforan) [reply src/Entities/Api/ReferenceData/OntologyVersion.cs:31]: it should be required. How can you have an OntologyVersion that doesn't relate to an Ontology?

better than breaking the model to suit your data would be fixing your data to match the model ;)

In this case, why not insert the only Ontology, and version, we support today, to fulfill the API's expectations.

As per the current QA API database:

Ontology: SNOMED-CT
Ontology Version: Value - 25

File: src/Directory/DataSeed/Services/SeedingService.cs [hunk 2]
```diff
@@ -22,209 +23,193 @@ namespace Biobanks.DataSeed.Services
...
+            return _db.Configs.FirstOrDefault(x => x.Key == key).Value;
+        }
+
+        private IEnumerable<T> ReadCsv<T>(string filePath = "") where T : class
+        {
+            if (string.IsNullOrEmpty(filePath))
             {
-                IgnoreReferences = true
-            };
+                filePath = $"{ _dataDir }/{ typeof(T).Name }.csv";
```
Comment (beforan) [inline src/Directory/DataSeed/Services/SeedingService.cs:205]: use `Path.Combine()`

File: src/Entities/Api/ReferenceData/OntologyVersion.cs (hunk 1 above)
Comment (beforan) [reply src/Entities/Api/ReferenceData/OntologyVersion.cs:31]: I see we have data for those, so I'm not sure what the issue is? when are we trying to insert null FKs?

**Review by beforan — CHANGES_REQUESTED** [review]

**Review by beforan — APPROVED** [review]

**Review by JosephLavagna — APPROVED** [review]

**Pull request merged**